src/commands/lightning/preview/app.ts

```diff
@@ -1,41 +1,137 @@
+    } else if (flags['device-type'] === Platform.android) {
+      await this.androidPreview(logger, flags);
+    }
+  }
+
+  private async desktopPreview(logger: Logger, flags: LightningPreviewAppFlags): Promise<void> {
+    const appName = flags['name'];
+    const targetOrg = flags['target-org'];
+    const platform = flags['device-type'];
+
+    logger.debug('Determining Local Dev Server url');
+    // todo: figure out how to make the port dynamic instead of hard-coded value here
+    const ldpServerUrl = PreviewUtils.generateWebSocketUrlForLocalDevServer(platform, '8081');
+    logger.debug(`Local Dev Server url is ${ldpServerUrl}`);
+
+    // appPath will resolve to one of the following:
+    //
+    //   lightning/app/<appId> => when the user is targetting a specific LEX app
+    //               lightning => when the user is not targetting a specific LEX app
+    //
+    let appPath = '';
+    if (appName) {
+      logger.debug(`Determining App Id for ${appName}`);
+
+      // The appName is optional but if the user did provide an appName then it must be
+      // a valid one.... meaning that it should resolve to a valid appId.
+      const appId = await OrgUtils.getAppId(targetOrg.getConnection(undefined), appName);
+      if (!appId) {
+        throw new Error(messages.getMessage('error.fetching.app-id', [appName]));
+      }
+
+      logger.debug(`App Id is ${appId}`);
+
+      appPath = `lightning/app/${appId}`;
+    } else {
+      logger.debug('No Lightning Experience application name provided.... using the default app instead.');
+      appPath = 'lightning';
+    }
+
+    // we prepend a '0.' to all of the params to ensure they will persist across redirects
+    const orgOpenCommandArgs = ['--path', `${appPath}?0.aura.ldpServerUrl=${ldpServerUrl}&0.aura.mode=DEVPREVIEW`];
+
+    // There are various ways to pass in a target org (as an alias, as a username, etc).
+    // We could have LightningPreviewApp parse its --target-org flag which will be resolved
+    // to an Org object (see https://github.com/forcedotcom/sfdx-core/blob/main/src/org/org.ts)
+    // then write a bunch of code to look at this Org object to try to determine whether
+    // it was initialized using Alis, Username, etc. and get a string representation of the
+    // org to be forwarded to OrgOpenCommand.
+    //
+    // Or we could simply look at the raw arguments passed to the LightningPreviewApp command,
+    // find the raw value for --target-org flag and forward that raw value to OrgOpenCommand.
+    // The OrgOpenCommand will then parse the raw value automatically. If the value is
+    // valid then OrgOpenCommand will consume it and continue. And if the value is invalid then
+    // OrgOpenCommand simply throws an error which will get bubbled up to LightningPreviewApp.
+    //
+    // Here we've chosen the second approach
+    const idx = this.argv.findIndex((item) => item.toLowerCase() === '-o' || item.toLowerCase() === '--target-org');
```

Comment:
Does it require a lot of parsing of the `Org` data? Or does an `Org` get fleshed out with all of its appropriate configuration data, once its parameter input value has been resolved to an object?

If it's the latter, then we could simply use `org.getUsername()` on the resolved org, and use that as the value to be passed to `org open`.

Reply:
`org.getUsername()` alone is not enough since it returns `string | undefined` and in case it returns undefined then we should look into pulling out the org alias name. The alias is available on `org.options` but the problem is that `options` is private. I could work around it by using `(org as any).options.aliasOrUsername` but there are 2 issues:

1. It's hacky and can break in the future if the fields of `options` change
2. `options.aliasOrUsername` is also nullable

Ideally we would do a contribution to `@salesforce/core` and add a method to `Org` that would have the logic of resolving the org identity for us. But for now I resorted to the simplest approach. There is plenty more work to be done in this repo and on the `preview app` command so we can always revisit and improve this if needed.

Reply:
Oh yeah, if you basically _don't_ get a fully configured Org data object by virtue of passing in the seeding value, then I agree: let's just go with the approach you have. My hope was that all got fleshed out for you behind the scenes, but sounds like no.

Reply:
One way or another we should be deferring the org parsing logic to the CLI functions we're calling. That shouldn't be this plugin's process to implement.